### PR TITLE
BLO-771 feat: 2fa validation and error states

### PR DESCRIPTION
**Pull request by simonheys**
### Issue / feature description

Adds validation, tests and error states. Fixes an issue with trying to re-add an account that was already in the backend. 

### Changes

- check existing shield-enabled accounts when user email returns no accounts
- refactor validation and add test cases
- add error screens and Storybook

### Checklist

- [x] Rebased to the last commit of the target branch (or merged)
- [x] Code self-reviewed
- [x] Code self-tested
- [x] Tests updated (if needed)
- [x] All tests are passing locally

### CODE REVIEW

File: packages/extension/src/ui/features/shield/ShieldValidationErrorScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        Please enter another email address to continue.
+      </>
+    ) : error === SHIELD_EMAIL_VALIDATION_FAILURE_SCENARIO_2 ? (
+      <>
+        Please use the same email address that you used to add Argent Shield to
+        your other accounts
+      </>
+    ) : (
+      <>
+        This address is associated with accounts from another seedphrase.
```
Comment (diegodelrieu) [inline packages/extension/src/ui/features/shield/ShieldValidationErrorScreen.tsx:45]: Not sure if this is intentional but this is the same message twice even though it seems like it's two different errors

**Review by diegodelrieu — APPROVED** [review]
Very nice refacto / tests

File: packages/extension/src/ui/features/shield/ShieldValidationErrorScreen.tsx (hunk 1 above)
Comment (simonheys) [reply packages/extension/src/ui/features/shield/ShieldValidationErrorScreen.tsx:45]: It's intentional, the difference is in one scenario user has no 2FA locally, and the other they have at least one 2FA account locally (I think...)

File: packages/storybook/src/features/shield/ShieldValidationErrorScreen.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+const Template: ComponentStory<typeof ShieldValidationErrorScreen> = (
+  props,
+) => (
+  <MemoryRouter initialEntries={["/"]}>
+    <ShieldValidationErrorScreen {...props} />
+  </MemoryRouter>
+)
+
+const onDone = () => console.log("onDone")
```
Comment (bluecco) [inline packages/storybook/src/features/shield/ShieldValidationErrorScreen.stories.tsx:23]: placeholder?

Comment (simonheys) [reply packages/storybook/src/features/shield/ShieldValidationErrorScreen.stories.tsx:23]: Intentional - this is just in Storybook, as a convenience logs 'onDone' to console when you click the button as an indication that it is working

**Review by janek26 — APPROVED** [review]

**Pull request merged**